agent/pool/pool.go

```diff
@@ -611,8 +570,12 @@ func (p *ConnPool) RPC(
 // AutoEncrypt.Sign is a one-off call and it doesn't make sense to pool that
 // connection if it is not being reused.
 func (p *ConnPool) rpcInsecure(dc string, nodeName string, addr net.Addr, method string, args interface{}, reply interface{}) error {
+	if dc != p.Datacenter {
+		return fmt.Errorf("insecure dialing prohibited between datacenters")
+	}
+
 	var codec rpc.ClientCodec
-	conn, _, err := p.DialTimeoutInsecure(dc, nodeName, addr, 1*time.Second, p.TLSConfigurator.OutgoingRPCWrapper())
+	conn, _, err := p.dial(dc, nodeName, addr, 1*time.Second, 0, RPCTLSInsecure)
```

Comment:
The second to last arg is zero because it doesn't matter what the value is, right? In the prior invocation RPCTLSInsecure was passed for both values (and the second-to-last value was still similarly ignored).

Reply:
Yes! `dial` ignores the second to last value if it is in `RPCTLSInsecure` mode: https://github.com/hashicorp/consul/pull/7966/files#diff-68d0f29195d4f1c4b73a26e8ddbbfbe8L417-L422.